sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2525,7 +2525,8 @@ class Analyzer(
     override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
```

Comment:
We might need to update the description for `ResolveOutputRelation` and refine "what's a safe cast?".

Reply:
btw, we might need to move some parts of this description to `TableOutputResolver`.

Reply:
I have removed the word "safe" in the comment. I think the comment "Detect plans that are not compatible with the output table and throw AnalysisException" already states that there is type coercion check in the rule.